sql/core/src/test/scala/org/apache/spark/sql/execution/WholeStageCodegenSuite.scala

```diff
@@ -165,6 +166,26 @@ class WholeStageCodegenSuite extends QueryTest with SharedSparkSession
     }
   }
 
+  test("SPARK-: Avoid spill in partial aggregation " +
+    "when spark.sql.aggregate.spill.partialaggregate.disabled is set") {
+    withSQLConf((SQLConf.SPILL_PARTIAL_AGGREGATE_DISABLED.key, "true"),
```

Comment:
This is not sufficient since the rows still get added to `UnsafeFixedWidthAggregationMap` , need to figure out a way to avoid adding elements to `UnsafeFixedWidthAggregationMap`. Please advise @cloud-fan  @gatorsmile @maropu

Reply:
Could you show us performance numbers?

Reply:
@maropu  I figured out few more improvements taht can be made to the generated code, I will test them and also addd the benchmark number.
Adding **WIP** tag to the title.

Reply:
@maropu 
I have added the benchmark to the description.
I have also figured out a way to add UT for this.
@maropu  @cloud-fan  @gatorsmile Can you please help review the change